### ISSUE TITLE
[Bug]: Silent refresh fails using localStorage when multiple tabs trying to refresh at the same time

### ISSUE THREAD

**Issue by Coldfen** [damienbod/angular-auth-oidc-client#1838]
### Version

16.0.0

### Please provide the exception or error you saw

```true
myConfigId - silent renew failed! Error: Error: authorizedCallback, token(s) validation failed, resetting.
```

### Steps to reproduce the behavior

```true
1) Set up OIDC Code Flow PKCE using refresh tokens (used 'ng add angular-auth-oidc-client')

2) Set up custom storage using localStorage (used example from https://angular-auth-oidc-client.com/docs/documentation/custom-storage)

3) To speed up testing, setup your authority to have short lived access and refresh tokens. (i.e. 30 - 50 seconds) 

4) Login using one browser [chrome] tab (i.e. this.oidcSecurityService.authorize()). Open console and verify refresh is working

5) Open a few more tabs. Open console and verify they are authorized and refreshing (in my case i open 6 or more)

6) Wait until error occurs. It shouldn't take long - usually 2-20 minutes with 30 - 40 seconds tokens. Open more tabs if needed.
```

### A clear and concise description of what you expected to happen.

```true
Access token should refresh successfully and all tabs remain authorized
```

### Additional context

The same problem was described in [#1662](https://github.com/damienbod/angular-auth-oidc-client/issues/1662)

![image](https://github.com/damienbod/angular-auth-oidc-client/assets/94360746/c977666a-5176-4937-98dd-4b9be9151020)

A temporary solution does not suit me well, because my token functions for 300 seconds.

My config: 
![image](https://github.com/damienbod/angular-auth-oidc-client/assets/94360746/4856e769-9348-4c32-9041-af26a1c5e1fb)

AuthConfigModule added to the imports of the main AppModule.

**Comment by joewIST** [comment]
Experiencing the same issue. Any help appreciated.

**Comment by CraigGP1** [comment]
Same issue.

**Comment by Ghostbird** [comment]
No, not as far as I know. The trick mentioned in #1662 is something I'm going to try, where the silent renew offset gets a random number. That won't fix it, but may significantly reduce the problem.

**Comment by joewIST** [comment]
Good luck. We tried that in the past but it didn't fix our issue.

**Comment by joewIST** [comment]
We tried that but unfortunately it doesn't resolve our issue.

**Comment by lemonCMS** [comment]
Is there any progress on this issue?

**Comment by FabianGosebrink** [comment]
The cleanest solution would be to queue up the refreshes to guarantee a sequential order🤔🤔🤔

**Comment by sebastiandenis** [comment]
I have the same issue :/

**Comment by Ghostbird** [comment]
@FabianGosebrink I think most proposed workarounds are trying to achieve this. I tried a random `renewTimeBeforeTokenExpiresInSeconds` for example, assuming that it would sequence the refreshes, but that didn't work.

**Comment by sebastiandenis** [comment]
I'm trying to make it not so painful for our users, who often open 10-30 tabs and work on them. The random renewTimeBeforeTokenExpiresInSeconds solution doesn't work, and it is even worse. 

When this happens, the lib sets the authResult value in the local storage to undefined. When I try to prevent it (using AbstractSecurityStorage implementation) - it is much better. One of the tabs crashes, but the other works fine. 

But it has deeper consequences inside the lib - for example, for the crashed instance of the tab, the silent renew process stops.

Not a clean way to deal with the issue, but I'm trying anything that could work for our users.

**Comment by sebastiandenis** [comment]
@FabianGosebrink is there any chance to fix this anytime soon? We've tried many workarounds, but it seems that you can't do too much from the outside of the lib. When this error occurs, the auth data is being reset in the local storage and it affects all the tabs.

**Comment by FabianGosebrink** [comment]
I would love to but lacking of time :( As soon as I got time I can try to fix this.

**Comment by jugarpeupv** [comment]
Hi @FabianGosebrink, could you give us a hint in how you think this should be implemented?

I was looking at the implementation suggested on the manfredsteyer/angular-oauth2-oidc using the Lock API

https://github.com/manfredsteyer/angular-oauth2-oidc/pull/1423

Trying to replicate it on this library, but i am currently stuck on linking this library into my application locally with this error. I don't have a clue why this might be happening

<img width="494" alt="image" src="https://github.com/user-attachments/assets/c43fcf9a-09cf-4223-b986-86ddc518d86d">

Anyway, if you could share your thoughts on this approach, thank you in advanced ^^

**Comment by joewIST** [comment]
Hi, has anyone found a reliable workaround for this? We are still stuck with the issue on v18. @FabianGosebrink Is this something that is planned in the near future? Thanks again for your help.

**Comment by skatterwe** [comment]
@jugarpeupv do you have the things you've implemented pushed to a branch and if so can you share it? maybe i can have a look. we have the same issue
